### PR TITLE
Implement Memory Behaviors

### PR DESCRIPTION
**Pull request by ryzokuken**
🎉

@pdurbin @jywarren

### CODE REVIEW

File: src/models/memory-behavior.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// TODO(@ryzokuken): Switch to a more persistent memory model later
+// (Hint: Databases)
+let memory = [];
+
+class MemoryBehavior extends Behavior {
+  constructor(trigger, action, keyword) {
+    const newAction = (botNick, username, ...args) => {
+      if (!utils.contains(memory, username)) {
+        memory.push(username);
```
Comment (jywarren) [inline src/models/memory-behavior.js]: Could we use a key value store here like `memory.get('key')` and `memory.set('key', 'value')`? That way we could also pass in an option for a function of what to remember - we could remember for example what time they were first remembered, or other things we haven't thought of yet. It seems more scalable to problems we want to solve soon like "responding only i to newcomers" and is closer to a key/value database model anyways. What do you think?

Comment (jywarren) [reply src/models/memory-behavior.js]: Later implementing db w something like this:

https://preview.npmjs.com/package/key-value-store

Comment (jywarren) [reply src/models/memory-behavior.js]: Ok or maybe https://www.npmjs.com/package/interface-datastore

Comment (ryzokuken) [reply src/models/memory-behavior.js]: @jywarren you up for a really quick chat?

**Comment by jywarren** [comment]
Oh hi, sorry I wasn't available! That sounds great, go ahead!

**Comment by jywarren** [comment]
Thanks Ujjwal!

**Comment by ryzokuken** [comment]
Thanks, Jeff! Merging this.

**Pull request merged**